[DOCS-12425] Adding Syslog setup to NDM Setup docs (#32553)
DataDog/documentation: content/en/network_monitoring/devices/syslog.md

---
title: Syslog 
further_reading:
- link: "/network_monitoring/devices/profiles"
  tag: "Documentation"
  text: "Using Profiles with Network Device Monitoring"
- link: "https://www.datadoghq.com/knowledge-center/network-monitoring/snmp-monitoring/"
  tag: "Knowledge Center"
  text: "SNMP Monitoring Overview"
---

## Overview

Network Device Monitoring (NDM) uses Syslog to provide visibility into the health and performance of your network infrastructure. By integrating your network devices with Datadog through Syslog, you can collect and analyze log data, monitor device behavior, troubleshoot issues, and maintain network stability.

> {{< img src="network_device_monitoring/syslog/ndm_syslog.png" alt="Network Device Monitoring side panel, highlighting the Syslog tab." style="width:100%;" >}}

Reviewer: The Syslogs tab is hard to pick out in such a large screenshot! Might be worth cropping it down to make it a little easier to find?

## Prerequisites

* Datadog Agent version 7.57 or later installed and running on a host that can receive Syslog messages from your network devices.

> * Network devices configured to send [Syslog messages][1] either directly to the Datadog Agent or through a proxy that forwards messages to the Datadog Agent.

Reviewer: ```suggestion
* Network devices configured to send [Syslog messages][1] either directly to the Datadog Agent, or through a proxy that forwards messages to the Datadog Agent.
```

## Configuration

{{< tabs >}}
{{% tab "Linux" %}}

1. Ensure the following settings are enabled in your `/etc/datadog-agent/datadog.yaml` file:

   ```yaml
   logs_enabled: true # enable logs collection
   logs_config:
     use_sourcehost_tag: true # adds a source_host tags to logs with the source IP
   ```

2. Create a Syslog listener configuration:

   - In `/etc/datadog-agent/conf.d/`, create a directory called `syslog.d/`.

>    - Inside `syslog.d/`, create a file named `conf.yaml` with the following:

Reviewer: ```suggestion
   - In `syslog.d/`, create a file named `conf.yaml` with the following:
```

   ```yaml
   logs:
     - type: syslog
       port: 514
       protocol: udp
       source: syslog
       service: <service> # optional tag
   ```

   * **`port`**: The port the Agent listens on for Syslog messages. Default is 514.
   * **`protocol`**: Set to `udp` or `tcp` based on your device configuration.
   * **`source`**: Custom source name for these logs in Datadog. Use `syslog` to correlate with NDM devices.

>    * **`service`**: Optional service name for unified service tagging.

Reviewer: These look like they should maybe be nested under the previous bullet point too?
```suggestion
      - **`port`**: The port the Agent listens on for Syslog messages. Default is 514.
      - **`protocol`**: Set to `udp` or `tcp` based on your device configuration.
      - **`source`**: Custom source name for these logs in Datadog. Use `syslog` to correlate with NDM devices.
      - **`service`**: Optional service name for unified service tagging.
```

   - Inside `syslog.d/`, create a file named `conf.yaml` with the following:

   ```yaml
   logs:
     - type: syslog
       port: 514
       protocol: udp
       source: syslog
       service: <service> # optional tag
   ```

   * **`port`**: The port the Agent listens on for Syslog messages. Default is 514.
   * **`protocol`**: Set to `udp` or `tcp` based on your device configuration.
   * **`source`**: Custom source name for these logs in Datadog. Use `syslog` to correlate with NDM devices.
   * **`service`**: Optional service name for unified service tagging.

3. [Restart the Datadog Agent][101] service from the Services console to apply the changes.

[101]: /agent/configuration/agent-commands/#restart-the-agent

{{% /tab %}}
{{% tab "Docker" %}}

1. Ensure the following settings are enabled in your `/etc/datadog-agent/datadog.yaml` file in your Docker volume:

   ```yaml
   logs_enabled: true # enable logs collection
   logs_config:
   use_sourcehost_tag: true # adds a source_host tags to logs with the source IP
   ```

2. Create a Syslog listener configuration:

   - Mount a volume to `/etc/datadog-agent/conf.d/` and create a directory called `syslog.d/`.

>    - Inside `syslog.d/`, create a file named `conf.yaml` with the following:

Reviewer: ```suggestion
   - In `syslog.d/`, create a file named `conf.yaml` with the following:
```

   ```yaml
   logs:
     - type: syslog
       port: 514
       protocol: udp
       source: syslog
       service: <service> # optional tag
   ```

   * **`port`**: The port the Agent listens on for Syslog messages. Default is 514.
   * **`protocol`**: Set to `udp` or `tcp` based on your device configuration.
   * **`source`**: Custom source name for these logs in Datadog. Use `syslog` to correlate with NDM devices.
   * **`service`**: Optional service name for unified service tagging.

3. Restart the [Docker container][101] to apply the changes.

[101]: /agent/configuration/agent-commands/#restart-the-agent

{{% /tab %}}
{{< /tabs >}}

### Log parsing

> The above steps send your Syslog data to Datadog. To ensure NDM correctly associates these logs with your monitored network devices, set up a custom log parsing pipeline that populates the `syslog_ip` tag with each device's source IP address.

Reviewer: Not sure if this is fully technically correct - I was just a little concerned that users might misunderstand and think that the config steps caused a one-time migration of data to Datadog rather than an ongoing pipeline.
```suggestion
After you complete the above steps, Syslog can start sending log data to Datadog. To ensure NDM correctly associates these logs with your monitored network devices, set up a custom log parsing pipeline that populates the `syslog_ip` tag with each device's source IP address.
```

   * **Source of the tag:** `source_host`
   * **Target tag:** `syslog_ip`

#### Scenario B: Using a Syslog proxy

When network devices send Syslog through a proxy (such as rsyslog or syslog-ng), the `source_host` tag reflects the proxy's IP address instead of the original device's IP.

To create the `syslog_ip` tag:

1. **Configure the proxy** to include the original source IP in the Syslog message payload.
   * **rsyslog example:** Use a template like `$template CustomFormat,"%fromhost-ip% %msg%\n"` to prepend the source IP to each message.
2. **Create a Log Processing Pipeline** to extract and map the IP address:
   1. In Datadog, navigate to [**Logs > Log Configuration > Pipelines**][3].
   2. Create a pipeline or select an existing one.
   3. Add a **Grok Parser** processor to extract the IP address from the message into a temporary attribute (for example, `@temp_ip`).
   4. Add a **Log Remapper** processor with the following configuration:
      * **Source of the attribute:** `@temp_ip` (or your chosen attribute name)
      * **Target tag:** `syslog_ip`
      * **Preserve source attribute:** Uncheck this option to remove the temporary attribute

> For detailed instructions, see the [Datadog Log Pipelines documentation][2].

Reviewer: ```suggestion
For more information, see the [Datadog Log Pipelines documentation][2].
```
What you gave is plenty detailed 🙂

## Verify Syslog messages

After your network devices are configured and the Datadog Agent is running, you can verify that Syslog data is being collected and sent to Datadog:

1. Navigate to the **[Log Explorer][4]** in your Datadog account.
2. **Filter by `source:syslog`** (or whatever source you specified in your `conf.yaml` file).

> 3. You should see your network device Syslog messages appearing in the Log Explorer.

Reviewer: I would incorporate this into the previous step; it's not really an action, but a consequence of the previous action, so I don't think it needs to be its own step.

> 4. **Verify `syslog_ip`:** Ensure that the `syslog_ip` tag is present and correctly populated with the network device's IP address for each relevant log entry.

Reviewer: This bolding seems a little different from what we usually do. I don't think it's necessarily a bad thing, but when there's a lot of bold in a procedure, it can be hard to figure out why or where you're supposed to focus. I think most of the time, we tend to limit bolding to screen elements that you're asking a user to interact with, but you and your PMs may decide to put emphasis in different places. Thought I'd bring it up anyway, just in case!

>    {{< img src="network_device_monitoring/syslog/log_syslog_ip.png" alt="Log explorer, filtering by `source:syslog`, highlighting the `syslog_ip` tag on the side panel." style="width:100%;" >}}

Reviewer: There's a lot going on in this screenshot! I might consider cropping it down a bit so there's more of a focus on the relevant part of the screen. And as a bonus, I think that would also help you avoid future maintenance if there's, say, an update to the top nav again!
